app/Model/Infrastructure/Repositories/CandidateRepository.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Model\Infrastructure\Repositories;
+
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\OptimisticLockException;
+use Doctrine\ORM\ORMException;
+use Model\Candidate\Candidate;
+use Model\Candidate\CandidateFunction;
+use Model\Candidate\FunctionNotFound;
+use Model\Candidate\Repositories\ICandidateRepository;
+use Model\DTO\Candidate\SkautisCandidate;
+use function assert;
+
+final class CandidateRepository extends AggregateRepository implements ICandidateRepository
+{
+    /**
+     * @param SkautisCandidate[] $candidates
+     *
+     * @throws ORMException
+     * @throws OptimisticLockException
+     */
+    public function saveCandidates(array $candidates) : void
+    {
+        $fillCandidateWith = [];
+        $candidateObjs     = [];
+        foreach ($candidates as $candidate) {
+            assert($candidate instanceof SkautisCandidate);
+            $candidateObjs[$candidate->getId()] = new Candidate(
+                $candidate->getId(),
+                $candidate->getPersonId(),
+                $candidate->getName(),
+                $this->getFunction($candidate->getFunctionId())
+            );
+            if (! $candidate->hasRunningMate()) {
+                continue;
+            }
+
+            $fillCandidateWith[$candidate->getId()] = $candidate->getRunningMateId();
+        }
+        // set relationship for candidate pairs
+        foreach ($fillCandidateWith as $primaryId => $secondaryId) {
+            $candidateObjs[$primaryId]->setCandidateWith($candidateObjs[$secondaryId]);
```

Comment:
Drobnost: Nemůžeme se tady zbavit té oboustrané vazby? Nestačí nám držet toho "hlavního" kandidáta a jeho spolukandidáta? Zbavili bysme se této obezličky `setCandidateWith` pro obejití té kruhové závislosti.

Reply:
tady to `setCandidateWith` je z toho důvodu, že když zakládám hlavního kandidáta, tak jeste nemám zalozeného spolu kandidáta, tedy na něj nemuzu tu vazbu udělat kdyz neexistuje.